pkg/core/config/template.go

```diff
@@ -59,7 +62,10 @@ func (t *Template) New(content []byte) ([]byte, error) {
 	// Deepmerge is not supported so a secrets override unencrypted values
 	templateValues := Merge(t.Values, t.Secrets)
 
-	tmpl, err := template.New("cfg").Funcs(funcMap).Parse(string(content))
+	tmpl, err := template.New("cfg").
+		Funcs(funcMap).
```

Comment:
I guess here order matter. I was wondering which one between sprig and updatecli map should have the priority. I am thinking in a long period where sprig introduce a function map that conflict with Updatecli.

Imho it's better to have updatecli function coming first and if a user in the futur want to use a new sprig function, then we'll revisit that decision.

Maybe all we need here is a comment

Reply:
Good catch. Thinking about it, I would actually put `Funcs(funcMap)` second, so that is always takes precedence if Sprig adds a function with the same name.

Reply:
Do you know how it is interpreted? Is the first or last one considered? 

The way I see it is if later on, sprig add a new function that conflict with Updatecli, then we can control how we deprecate adapt without breaking existing manifest

Because if sprig is used first then all existing manifest will stop working if sprig introduce a function that conflict with Updatecli while on the other side, we can plan so transition without breaking existing manifest.

Or we consider we have good test coverage for updatecli function and we'll catch the error if we bump the sprig library version :trollface:

Reply:
Subsequent `Funcs` calls will override any existing functions:
https://github.com/golang/go/blob/master/src/text/template/template.go#L182

So if we add `Funcs(funcMap)` last, it will always take precedence. I agree that this is the safest option.

Reply:
Thanks for your reactivity